src/options/normalizeSQL.js

```diff
@@ -17,9 +17,20 @@ function normalizeDate (where) {
  * @param {Object} options - object that may contain 'fields' or 'outFields' property
  */
 function normalizeFields (options) {
-  const fields = options.fields || options.outFields
+  let fields = options.fields || options.outFields
+
+  // TODO: Clearer set of if statements
   if (fields === '*') return undefined
-  if (typeof fields === 'string' || fields instanceof String) return fields.split(',')
+  if (typeof fields === 'string' || fields instanceof String) fields = fields.split(',')
+  if (fields instanceof Array && options.toEsri) {
+    // If idField has been set for use as OBJECTID, we must replace OBJECTID with idField in the query fields
```

Comment:
is this a consequence of renaming the idField?

Reply:
Yes and no.  I think we need this regardless.  Some ArcGIS client requests appear to send only "OBJECTID" in the `outFields` parameter.  If an `idField` has been set in provider metadata, the `SELECT`s on such request won't select it the attribute denoted by `idField`, and it then won't be available for assignment to the OBJECTID value in the `esriFy` function.

Reply:
Are you sure this is not because we are setting metadata incorrectly?

We're not actually using this template anymore, but we should be:

https://github.com/koopjs/FeatureServer/blob/master/templates/oid-field.json